File: elixir/easy/1207/1207.exs
defmodule Solution do
  @spec unique_occurrences(arr :: [integer]) :: boolean
  def unique_occurrences(arr) do
    values =
      arr
      |> Enum.frequencies()
      |> Map.values()

    values == Enum.uniq(values)
  end
end

ExUnit.start()

defmodule AssertionTest do
  use ExUnit.Case, async: true

  test "test_1" do
    assert Solution.unique_occurrences([1, 2, 2, 1, 1, 3]) == true
  end

  test "test_2" do
    assert Solution.unique_occurrences([1, 2]) == false
  end
end

File: py/easy/1207/1207.py
from collections import Counter
from typing import List


class Solution:
    def uniqueOccurrences(self, arr: List[int]) -> bool:
        values = Counter(arr).values()
        return len(set(values)) == len(values)


solution = Solution()
assert solution.uniqueOccurrences([1, 2, 2, 1, 1, 3]) is True
assert solution.uniqueOccurrences([1, 2]) is False
